commands/api-gateway/gateway/customDomains/remove.go

```diff
@@ -0,0 +1,68 @@
+package customDomains
+
+import (
+	"context"
+	"fmt"
+	"github.com/ionos-cloud/ionosctl/v6/commands/api-gateway/completer"
+	"github.com/ionos-cloud/ionosctl/v6/internal/client"
+	"github.com/ionos-cloud/ionosctl/v6/internal/constants"
+	"github.com/ionos-cloud/ionosctl/v6/internal/core"
+	"github.com/ionos-cloud/sdk-go-bundle/products/apigateway/v2"
+	"github.com/spf13/viper"
+)
+
+func RemovetCmd() *core.Command {
+	cmd := core.NewCommand(context.Background(), nil, core.CommandBuilder{
+		Namespace: "apigateway",
+		Resource:  "customdomains",
+		Verb:      "remove",
+		Example:   "ionosctl apigateway gateway customdomains remove --gateway-id ID --custom-domains-id ID",
+		PreCmdRun: func(c *core.PreCommandConfig) error {
+			if err := core.CheckRequiredFlags(c.Command, c.NS, constants.FlagGatewayID, constants.FlagCustomDomainsId); err != nil {
+				return err
+			}
+
+			return nil
+		},
+		CmdRun: func(c *core.CommandConfig) error {
+			apigatewayId := viper.GetString(core.GetFlagName(c.NS, constants.FlagGatewayID))
+			customDomainsId := viper.GetInt(core.GetFlagName(c.NS, constants.FlagCustomDomainsId))
+			usedApiGateway, _, err := client.Must().Apigateway.APIGatewaysApi.ApigatewaysFindById(context.Background(), apigatewayId).Execute()
+			if err != nil {
+				return err
+			}
+			input := usedApiGateway.Properties
+			if input.CustomDomains == nil {
+				fmt.Errorf("There are no custom domains defined in this apigateway!")
+			} else if customDomainsId == 0 {
+				input.CustomDomains = input.CustomDomains[1:]
+			} else if customDomainsId < len(input.CustomDomains) {
+				input.CustomDomains = input.CustomDomains[:len(input.CustomDomains)-1]
+			} else if customDomainsId < 0 || customDomainsId >= len(input.CustomDomains) {
+				return fmt.Errorf("Invalid custom domain index")
+			} else {
+				input.CustomDomains = append(input.CustomDomains[:customDomainsId], input.CustomDomains[customDomainsId+1:]...)
+			}
```

Comment:
this is a similar case to `add` customdomains. The error is not returned, and the logic is brittle and can be reduced to an error (or two if you want to be fancy) check and an append line

```suggestion
if input.CustomDomains == nil || len(input.CustomDomains) == 0 {
	return fmt.Errorf("There are no custom domains defined in this API Gateway")
}

if customDomainsId < 0 || customDomainsId >= len(input.CustomDomains) {
	return fmt.Errorf("Invalid custom domain index")
}

input.CustomDomains = append(input.CustomDomains[:customDomainsId], input.CustomDomains[customDomainsId+1:]...)
```